test/unit_test/components/movement_sensor_test.dart

```diff
@@ -0,0 +1,285 @@
+import 'package:flutter_test/flutter_test.dart';
+import 'package:grpc/grpc.dart';
+import 'package:viam_sdk/src/components/movement_sensor/service.dart';
+import 'package:viam_sdk/src/gen/component/movementsensor/v1/movementsensor.pbgrpc.dart';
+import 'package:viam_sdk/src/resource/manager.dart';
+import 'package:viam_sdk/src/utils.dart';
+import 'package:viam_sdk/viam_sdk.dart';
+
+class FakeMovementSensor extends MovementSensor {
+  Map<String, dynamic>? extra;
+
+  @override
+  String name;
+
+  FakeMovementSensor(this.name);
+
+  @override
+  Future<Map<String, dynamic>> doCommand(Map<String, dynamic>? command) async {
+    return {'command': command};
+  }
+
+  @override
+  Future<Map<String, double>> accuracy({Map<String, dynamic>? extra}) async {
+    this.extra = extra;
+    return {'accuracy': 0};
+  }
+
+  @override
+  Future<Vector3> angularVelocity({Map<String, dynamic>? extra}) async {
+    this.extra = extra;
+    return Vector3(x: 0);
+  }
+
+  @override
+  Future<double> compassHeading({Map<String, dynamic>? extra}) async {
+    this.extra = extra;
+    return 0;
+  }
+
+  @override
+  Future<Vector3> linearAcceleration({Map<String, dynamic>? extra}) async {
+    this.extra = extra;
+    return Vector3(x: 0);
+  }
+
+  @override
+  Future<Vector3> linearVelocity({Map<String, dynamic>? extra}) async {
+    this.extra = extra;
+    return Vector3(x: 0);
+  }
+
+  @override
+  Future<Orientation> orientation({Map<String, dynamic>? extra}) async {
+    this.extra = extra;
+    return Orientation(oX: 0);
+  }
+
+  @override
+  Future<Position> position({Map<String, dynamic>? extra}) async {
+    this.extra = extra;
+    return Position(GeoPoint(latitude: 0), 0);
+  }
+
+  @override
+  Future<Properties> properties({Map<String, dynamic>? extra}) async {
+    this.extra = extra;
+    return Properties(angularVelocitySupported: false);
+  }
+}
+
+void main() {
+  group('FakeMovementSensor Tests', () {
+    const String name = 'movementSensor';
+    late FakeMovementSensor movementSensor;
```

Comment:
https://dart.dev/null-safety/understanding-null-safety#late-variables

> To handle the common pattern of state with delayed initialization, we’ve added a new modifier, `late`.

I am declaring the variable `movementSensor`, but it will be initialized later in the code, so i mark it as `late`.